platforms/nuttx/src/px4/microchip/mpfs/hrt/hrt.c

```diff
@@ -180,20 +190,54 @@ hrt_tim_isr(int irq, void *context, void *arg)
 }
 
 
+/**
+ * fumction to retrieve current timestamp in timer resolution.
+ * this handles the possible overflows and returns a monotonically
+ * increasing time
+ */
+
+inline static uint64_t hrt_get_current_time(void)
+{
+	uint64_t	curr_time;
+	static uint64_t	prev_curr_time;
+
+	/* get the current counter value */
+	uint32_t count = getreg32(MPFS_MSTIMER_LO_BASE + MPFS_MSTIMER_TIM1VALUE_OFFSET);
+
+	curr_time = base_time + (loadval - count);
+
+	/* This check can handle all the timer wraparound cases */
+	if (prev_curr_time > curr_time) {
```

Comment:
This is fishy; for some reason it seems to work though. I will study it more; it should somehow use the previous method of storing the previous "count" and comparing to that.